common/rdm/PidStoreLoader.cpp

```diff
@@ -127,126 +121,168 @@ const RootPidStore *PidStoreLoader::LoadFromStream(std::istream *data,
+ */
+bool PidStoreLoader::LoadFromProto(ManufacturerMap *pid_data,
+                                   const ola::rdm::pid::PidStore &proto,
+                                   bool validate) {
+  set<uint16_t> seen_manufacturer_ids;
+
+  ManufacturerMap::iterator iter = STLLookupOrInsertNew(
+      pid_data, ESTA_MANUFACTURER_ID);
+  if (!GetPidList(iter->second, proto, validate, true)) {
+    return false;
+  }
 
+  for (int i = 0; i < proto.manufacturer_size(); ++i) {
+    const ola::rdm::pid::Manufacturer &manufacturer = proto.manufacturer(i);
 
-/**
- * Get a list of pids from a protobuf object.
+    if (STLContains(seen_manufacturer_ids, manufacturer.manufacturer_id())) {
+      OLA_WARN << "Manufacturer id " << manufacturer.manufacturer_id() <<
+          "(" << manufacturer.manufacturer_name() <<
+          ") listed more than once in the PIDs file";
+      return false;
+    }
+    seen_manufacturer_ids.insert(manufacturer.manufacturer_id());
+
+    ManufacturerMap::iterator iter = STLLookupOrInsertNew(
+        pid_data, manufacturer.manufacturer_id());
+    if (!GetPidList(iter->second, manufacturer, validate, false)) {
+      return false;
+    }
+  }
+
+  return true;
+}
+
+/*
+ * @brief Populate a PidMap from a protobuf object that has a set of repeated
+ * PIDs.
  */
 template <typename pb_object>
-bool PidStoreLoader::GetPidList(vector<const PidDescriptor*> *pids,
+bool PidStoreLoader::GetPidList(PidMap *pid_map,
                                 const pb_object &store,
                                 bool validate,
                                 bool limit_pid_values) {
-  set<uint16_t> seen_values;
+  set<uint16_t> seen_pids;
   set<string> seen_names;
-  bool ok = true;
 
   for (int i = 0; i < store.pid_size(); ++i) {
     const ola::rdm::pid::Pid &pid = store.pid(i);
 
     OLA_DEBUG << "Loading " << pid.name();
     if (validate) {
-      set<uint16_t>::const_iterator value_iter = seen_values.find(pid.value());
-      if (value_iter != seen_values.end()) {
-        OLA_WARN << "Pid " << pid.value() << " exists multiple times in the "
-            "pid file";
-        ok = false;
-        break;
+      if (STLContains(seen_pids, pid.value())) {
+        OLA_WARN << "Pid " << pid.value()
+                 << " exists multiple times in the pid file";
+        return false;
       }
-      seen_values.insert(pid.value());
-
-      set<string>::const_iterator name_iter = seen_names.find(pid.name());
-      if (name_iter != seen_names.end()) {
-        OLA_WARN << "Pid " << pid.name() << " exists multiple times in the "
-            "pid file";
-        ok = false;
-        break;
+      seen_pids.insert(pid.value());
+
+      if (STLContains(seen_names, pid.name())) {
+        OLA_WARN << "Pid " << pid.name()
+                 << " exists multiple times in the pid file";
+        return false;
       }
       seen_names.insert(pid.name());
 
       if (limit_pid_values && pid.value() > 0x8000 && pid.value() < 0xffe0) {
```

Comment:
Are these not defined already, if not we should make them constants. Or add a pid.isvalid()